src/Components/WebAssembly/WebAssembly/src/Hosting/WebAssemblyHostBuilder.cs

```diff
@@ -187,6 +187,7 @@ public WebAssemblyHost Build()
         internal void InitializeDefaultServices()
         {
             Services.AddSingleton<IJSRuntime>(DefaultWebAssemblyJSRuntime.Instance);
+            Services.AddSingleton<IJSUnmarshalledRuntime>(DefaultWebAssemblyJSRuntime.Instance);
```

Comment:
As per comment at https://github.com/dotnet/aspnetcore/pull/24640/files#r467819802, I recommend not having this extra registration.